### ISSUE TITLE
Closing non text editor with `CMD+W` cause quit prompt to appear

### ISSUE THREAD

**Issue by artdiniz** [artdiniz/quit-control-vscode#3]
VSCode only gives textEditorCount. Other editors aren't included and we can't tell when there are no tabs open.

**Comment by schmuli** [comment]
@artdiniz, thanks for this plugin, I've been bitten by this "feature" many times.

I just installed and ran into this issue. 

Do you have ideas how to fix this? 

Currently, Ctrl+W on the new Welcome tab or Extension details tab will trigger the prompt.

**Comment by artdiniz** [comment]
@schmuli If I could tell how many tabs are open in VSCode I could only show the prompt when there are no tabs open. What happens is that VSCode gives me this info but it doesn't count tabs like the Extension details tab or the Welcome tab in it. 

I was reading the [vscode namespace API](https://code.visualstudio.com/docs/extensionAPI/vscode-api) and thought of some possible workarounds. I'm not sure if it is possible so I will make some tests this weekend and update this thread if I find it is possible to do it =)

**Comment by artdiniz** [comment]
Partially fixed this in ab4e6eba195566563ac3b957f5db6b0f7bc23f5b. Prompt is still appearing when no text editors are open and `CMD + W` is pressed in a non-text tab.

**Comment by schmuli** [comment]
You might need to check for an existing issue for such an API, or even open a new issue, with VSCode.

**Comment by artdiniz** [comment]
Two years later...🙂

### TL;DR;
There is only one known  edge case where I didn't figured out a way to stop the prompt from showing up:
* When closing a extension description tab and there are 0 text editors and 0 or more extension description tabs.

___
The number of open tabs still can't be reliably counted because of VSCode API limitations described in this issue and [issue #8](https://github.com/artdiniz/quit-control-vscode/issues/8).

3e33eb2  fixed cases where it was showing up and there were visible or active text editors in window.

2ee2afe fixed cases where it was showing up after closing an extension description page. It didn't fixed all cases! It only fixed cases where we could reliably detect that an extension description tab was being closed. The code now explicitly says that close window prompt only shows up when we may have an empty window or an undetectable extension page.

**Comment by artdiniz** [comment]
3e33eb2 and 2ee2afe are being published in v3.1.0 today 🙂
